src/HW_models/NRF_EGU.c

```diff
@@ -4,10 +4,925 @@
+{
+	PPI_SIDEEFFECT_TASK(3, 15);
+}
+
+void nrf_egu_4_TASK_TRIGGER_0(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 0);
+}
+
+void nrf_egu_4_TASK_TRIGGER_1(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 1);
+}
+
+void nrf_egu_4_TASK_TRIGGER_2(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 2);
+}
+
+void nrf_egu_4_TASK_TRIGGER_3(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 3);
+}
+
+void nrf_egu_4_TASK_TRIGGER_4(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 4);
+}
+
+void nrf_egu_4_TASK_TRIGGER_5(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 5);
+}
+
+void nrf_egu_4_TASK_TRIGGER_6(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 6);
+}
+
+void nrf_egu_4_TASK_TRIGGER_7(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 7);
+}
+
+void nrf_egu_4_TASK_TRIGGER_8(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 8);
+}
+
+void nrf_egu_4_TASK_TRIGGER_9(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 9);
+}
+
+void nrf_egu_4_TASK_TRIGGER_10(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 10);
+}
+
+void nrf_egu_4_TASK_TRIGGER_11(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 11);
+}
+
+void nrf_egu_4_TASK_TRIGGER_12(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 12);
+}
+
+void nrf_egu_4_TASK_TRIGGER_13(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 13);
+}
+
+void nrf_egu_4_TASK_TRIGGER_14(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 14);
+}
+
+void nrf_egu_4_TASK_TRIGGER_15(void)
+{
+	PPI_SIDEEFFECT_TASK(4, 15);
+}
+
+void nrf_egu_5_TASK_TRIGGER_0(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 0);
+}
+
+void nrf_egu_5_TASK_TRIGGER_1(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 1);
+}
+
+void nrf_egu_5_TASK_TRIGGER_2(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 2);
+}
+
+void nrf_egu_5_TASK_TRIGGER_3(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 3);
+}
+
+void nrf_egu_5_TASK_TRIGGER_4(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 4);
+}
+
+void nrf_egu_5_TASK_TRIGGER_5(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 5);
+}
+
+void nrf_egu_5_TASK_TRIGGER_6(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 6);
+}
+
+void nrf_egu_5_TASK_TRIGGER_7(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 7);
+}
+
+void nrf_egu_5_TASK_TRIGGER_8(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 8);
+}
+
+void nrf_egu_5_TASK_TRIGGER_9(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 9);
+}
+
+void nrf_egu_5_TASK_TRIGGER_10(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 10);
+}
+
+void nrf_egu_5_TASK_TRIGGER_11(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 11);
+}
+
+void nrf_egu_5_TASK_TRIGGER_12(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 12);
+}
+
+void nrf_egu_5_TASK_TRIGGER_13(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 13);
+}
+
+void nrf_egu_5_TASK_TRIGGER_14(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 14);
+}
+
+void nrf_egu_5_TASK_TRIGGER_15(void)
+{
+	PPI_SIDEEFFECT_TASK(5, 15);
+}
+
+/**
+ * @brief Get Interrupt Id of corresponding EGU peripheral
+ *
+ * @param p_reg NRF_EGU_Type pointer
+ * @return IRQn for p_reg
+ */
+#define REG_TO_IRQ(p_reg)   REG_TO_ID(p_reg) + SWI0_EGU0_IRQn
+
+/**
+ * @brief Check if channel has enabled interrupts
+ *
+ * @param p_reg NRF_EGU_Type pointer
+ * @param channel uint8_t channel number to check
+ * @return True if interrupt is enabled for channel in p_reg, otherwise return False
+ */
+#define CHANNEL_INTERRUPT_FLAG(p_reg, channel)    (p_reg->INTEN & nrf_egu_channel_int_get(channel))
```

Comment:
The same is in line 138:
#define CHANNEL_INTERRUPT_FLAG(p_reg, channel)    (p_reg->INTEN & nrf_egu_channel_int_get(channel))

Reply:
removed duplication